system/security/shared/caching.cpp

```diff
@@ -538,11 +538,20 @@ void  CPermissionsCache::setCacheTimeout(int timeoutSeconds)
 
     if (timeoutSeconds)
     {
+        unsigned threadStartWaitTimeSeconds = 5;
+
+        // If the fill thread has not been started, start it
         if (!m_fileScopeCacheFillThread.joinable())
         {
             DBGLOG("CACHE: CPermissionsCache starting managedFilesScopeCacheFillThread, timeout = %d", timeoutSeconds);
             m_fileScopeCacheFillThread = std::thread(&CPermissionsCache::managedFileScopesCacheFillThread, this);
-            m_fileScopeCacheFillSem.wait(5000);   // wait for thread to start
+            m_fileScopeCacheFillSem.wait(threadStartWaitTimeSeconds * 1000);   // wait for thread to start
```

Comment:
how do you know 5 seconds is enough? How long will the thread take to complete populating on a busy big system with lots of managed scopes? Isn't the suspicion that it takes significant time. Incidentally, it would be good to produce some stats on how long spent there.

I suggested in my last comments, that if done like this, it should loop and warn, but this (blocking in approach  tor/setCacheTimeout) is far from ideal...
The query threads are the ones that should block if cache not valid really - as outlined by Gavin before : https://github.com/hpcc-systems/HPCC-Platform/pull/19673#discussion_r2028401623

But pragmatically, a loop with warning may be okay for now, revisit after.

Reply:
This code does not block until the cache has been filled. It only blocks until the fill thread has locked the cache for write. Once the write lock is acquired (which should be right away), the semaphore is signaled which should release the wait. 

Since the lock has been obtained, if we return from here (which now is part of construction) and a client thread were to try and read from the cache, it would block because the fill thread has acquired the lock and is doing the initial fill. This helps keep the read manages scopes code simpler while protecting against any early readers.

If this is not acceptable, then I can change.

Reply:
You’re right, I’d forgotten the signal was immediate. 5 seconds should be enough in that case, but still possible under extreme circumstances to take longer. e.g. some memory locking scenarios. It would be safer to loop and warn.